stats.js

```diff
@@ -2,21 +2,87 @@ const AQ = require('./gofish/aquarium.js');
-
-    await connect();
-    let operation_types = await AQ.OperationType.all();
-    
-    for ( let i=0; i<operation_types.length; i++ ) {
+    for (let i = 0; i < operation_types.length; i++) {
         let response = await AQ.get("/operation_types/"+operation_types[i].id+"/stats");
         let stats = JSON.parse(response.data);
         console.log(`${i}: ${operation_types[i].category}/${operation_types[i].name}`);
-        for(const [key,val] of Object.entries(stats)) {
+        for (const [key, val] of Object.entries(stats)) {
             console.log("\t" + key + ": " + val);
         }
     }
+}
+
+async function most_recent() {
+    let operation_types = await get_operation_types();
+    let stat_map = {};
+
+    for (let i = 0; i < operation_types.length; i++) {
+        let response = await AQ.get("/operation_types/"+operation_types[i].id+"/stats");
+        let stats = JSON.parse(response.data);
+        console.log(`Processing ${operation_types[i].category}/${operation_types[i].name}`);
+        if (stats.last_run) {
+            stat_map[i] = stats;
+        }
+    }
+
+    let sorted_stats = Object.keys(stat_map).map(function(key) {
+        return [key, stat_map[key]];
+    });
+
+    sorted_stats.sort(function (fst, snd) {
+        return snd[1].last_run.localeCompare(fst[1].last_run);
+    });
+
+    console.log("Retrieving top 10 most recent runs");
 
+    print_top_10(operation_types, sorted_stats);
+}
+
+async function most_done() {
+    let operation_types = await get_operation_types();
+    let stat_map = {};
+
+    for (let i = 0; i < operation_types.length; i++) {
+        let response = await AQ.get("/operation_types/"+operation_types[i].id+"/stats");
+        let stats = JSON.parse(response.data);
+        console.log(`Processing ${operation_types[i].category}/${operation_types[i].name}`);
+        if (stats.done) {
+            stat_map[i] = stats;
+        }
+    }
+
+    let sorted_stats = Object.keys(stat_map).map(function(key) {
+        return [key, stat_map[key]];
+    });
+
+    sorted_stats.sort(function (fst, snd) {
+        return snd[1].done - fst[1].done;
+    });
+
+    console.log("Retrieving top 10 most done runs");
+
+    print_top_10(operation_types, sorted_stats);
+}
+
+function print_top_10(operation_types, sorted_stats) {
+    for (let i = 0; i < 10; i++) {
+        let stats = sorted_stats[i];
+        let index = stats[0];
+        let stat_data = stats[1];
+        console.log(`${i + 1}: ${operation_types[index].category}/${operation_types[index].name}`);
+        for(const [key,val] of Object.entries(stat_data)) {
+            console.log("\t" + key + ": " + val);
+        }
+    }
+}
+
+async function get_operation_types () {
+    console.log("Retrieving operation type stats");
+    await connect();
+    let operation_types = await AQ.OperationType.all();
+    return operation_types;
 }
 
-module.exports = stats;
+module.exports = {stats, most_done, most_recent};
```

Comment:
How about defining a class Stats and exporting the Stats class. operation_types would be become this.operation_types so all your methods have access to it. You could put the stats into each this.operation_types[i] or you could have a this.stat_map[i]

Then in the crush main file you would do things like

stats = new Stats(); // the constructor would do what get_operation_types does plus what lines 21-28 do. 
                                  // or you could call stats.retrieve()

stats.sort_by_last_run()  // or
stats.sort_by_most_done() // or don't sort

stats.display(stats.all())      // all() just returns this.operation_types
stats.display(stats.top(10)) // top(n) would return this.operation_types.slice(0,n)



stats.all just returns this.operation_types
stats.most_recent(n) sorts the operation types by data and then returns the top n
etc.

Reply:
done!